uint8 FIRE = 0
uint8 MOVE = 1
uint8 TURN = 2
uint8 DIE = 3
uint8 action

uint8 MOVE_UP = 4
uint8 MOVE_LEFT = 5
uint8 MOVE_DOWN = 6
uint8 MOVE_RIGHT = 7
uint8 move_direction

uint8 TURN_ANGLE_0 = 8
uint8 TURN_ANGLE_45 = 9
uint8 TURN_ANGLE_90 = 10
uint8 TURN_ANGLE_135 = 11
uint8 TURN_ANGLE_180 = 12
uint8 TURN_ANGLE_225 = 13
uint8 TURN_ANGLE_270 = 14
uint8 TURN_ANGLE_315 = 15
uint8 turn_direction

int32 id